uefi-raw/src/protocol/device_path.rs

```diff
@@ -7,8 +63,8 @@ use crate::{guid, Char16, Guid};
 ///
 /// Note that the fields in this struct define the header at the start of each
 /// node; a device path is typically larger than these four bytes.
-#[derive(Debug)]
-#[repr(C)]
+#[derive(Copy,Clone,Debug)]
+#[repr(C,packed)]
```

Comment:
Please don't use `packed`. Experience shows that `packed` makes
it annoying to access members. Only use it if it is necessary.
Here, it doesn't seem to have an effect. Look:

```rust
pub struct Base {
    pub major_type: u8,
    pub sub_type: u8,
    pub length: [u8; 2],
}

pub struct Packed {
    pub major_type: u8,
    pub sub_type: u8,
    pub length: [u8; 2],
}

fn main() {
    assert_eq!(size_of::<Base>(), size_of::<Packed>())
}
```

I'm fine with a unit test that ensures the ABI/size_of, tho! We also do this in other parts of the crate I think